naff/models/discord/asset.py

```diff
@@ -52,6 +52,21 @@ def url(self) -> str:
         ext = ".gif" if self.animated else ".png"
         return f"{self._url}{ext}?size=4096"
 
+    def as_url(self, *, extension: str | None = None, size: int = 4096) -> str:
+        """
+        Get the url of this asset.
+
+        args:
+            extension: The extension to override the assets default with
+            size: The size of asset to return
+
+        returns:
+            A url for this asset with the given parameters
+        """
+        if not extension:
+            extension = ".gif" if self.animated else ".png"
+        return f"{self._url}{extension}?size={size}"
```

Comment:
Suggesting many people are going to do `extension = "jpg"` or the like (at least I did), it's probably worthwhile to make it add the dot before the extension automatically. A couple of lines would need to be changed, but it wouldn't be hard.

Reply:
Resolved in 8cea7bf95b3b4996f00eb919c32378480c02e795 to support both usages